src/ObjectUtils/CoffFile.cpp

```diff
@@ -38,82 +39,58 @@ class CoffFileImpl : public CoffFile {
-  // start of the section. To get the address relative to the start of the object
-  // file, we need to find the section and add the base address of the section.
-  llvm::object::COFFSymbolRef coff_symbol_ref = object_file_->getCOFFSymbol(symbol_ref);
-  int32_t section_id = coff_symbol_ref.getSectionNumber();
-  const llvm::object::coff_section* section = nullptr;
-  std::error_code err = object_file_->getSection(section_id, section);
-  if (err) {
-    return ErrorMessage(absl::StrFormat("Section of symbol not found: %s", err.message()));
+  const auto dwarf_context = llvm::DWARFContext::create(*owning_binary_.getBinary());
+  if (dwarf_context != nullptr) {
+    has_dwarf_debug_info_ = true;
   }
-  return section->VirtualAddress;
 }
 
-ErrorMessageOr<SymbolInfo> CoffFileImpl::CreateSymbolInfo(
-    const llvm::object::SymbolRef& symbol_ref) {
-  if ((symbol_ref.getFlags() & llvm::object::BasicSymbolRef::SF_Undefined) != 0) {
-    return ErrorMessage("Symbol is defined in another object file (SF_Undefined flag is set).");
-  }
-  const std::string name = symbol_ref.getName() ? symbol_ref.getName().get() : "";
-  // Unknown type - skip and generate a warning.
-  if (!symbol_ref.getType()) {
-    LOG("WARNING: Type is not set for symbol \"%s\" in \"%s\", skipping.", name,
-        file_path_.string());
-    return ErrorMessage(absl::StrFormat(R"(Type is not set for symbol "%s" in "%s", skipping.)",
-                                        name, file_path_.string()));
-  }
-  // Limit list of symbols to functions. Ignore sections and variables.
-  if (symbol_ref.getType().get() != llvm::object::SymbolRef::ST_Function) {
-    return ErrorMessage("Symbol is not a function.");
-  }
-
-  auto section_offset = GetSectionOffsetForSymbol(symbol_ref);
-  if (section_offset.has_error()) {
-    return ErrorMessage(absl::StrFormat("Error retrieving section offset for symbol \"%s\": %s",
-                                        name, section_offset.error().message()));
+static void FillDebugSymbolsFromDWARF(uint64_t image_base, llvm::DWARFContext* dwarf_context,
+                                      ModuleSymbols* module_symbols) {
+  for (const auto& info_section : dwarf_context->compile_units()) {
+    for (uint32_t index = 0; index < info_section->getNumDIEs(); ++index) {
+      llvm::DWARFDie full_die = info_section->getDIEAtIndex(index);
+      // We only want symbols of functions, which are DIEs with isSubprogramDIR()
+      // and not of inlined functions, which are DIEs with isSubroutineDIE().
+      if (full_die.isSubprogramDIE()) {
+        SymbolInfo symbol_info;
+        std::string name(full_die.getName(llvm::DINameKind::LinkageName));
+        symbol_info.set_name(name);
+        symbol_info.set_demangled_name(llvm::demangle(name));
+
+        uint64_t low_pc;
+        uint64_t high_pc;
+        uint64_t unused_section_index;
+        if (!full_die.getLowAndHighPC(low_pc, high_pc, unused_section_index)) {
+          continue;
+        }
```

Comment:
bit: move this before getName? I suspect there is some number of incomplete subprograms

Reply:
Done.